docs/source/deep_dives/comet_logging.rst

```diff
@@ -0,0 +1,49 @@
+.. _comet_logging:
+
+================
+Logging to Comet
+================
+
+This deep-dive will guide you through how to set up logging to Comet in torchtune.
+
+.. grid:: 1
+
+    .. grid-item-card:: :octicon:`mortar-board;1em;` What this deep-dive will cover
+
+      * How to get started with Comet
+      * How to use the :class:`~torchtune.utils.metric_logging.CometLogger`
+      * How to log configs, metrics, and model checkpoints to Comet
+
+Torchtune supports logging your training runs to `Comet <https://www.comet.com/site/?utm_source=torchtune&utm_medium=docs&utm_content=docs>`_.
+An example Comet workspace from a torchtune fine-tuning run can be seen in the screenshot below.
+
+.. image:: ../_static/img/comet_torchtune_project.png
+  :alt: torchtune workspace in Comet
+  :width: 100%
+  :align: center
+
+.. note::
+
+  You will need to install the :code:`comet_ml` package to use this feature.
+  You can install it via pip:
+
+  .. code-block:: bash
+    comet init --api-key
+Metric Logger
+-------------
+
+The only change you need to make is to add the metric logger to your config. Comet will log the metrics and model checkpoints for you.
+
+.. code-block:: yaml
+    # enable logging to the built-in CometLogger
+    metric_logger:
+      _component_: torchtune.utils.metric_logging.CometLogger
+      # the Comet project to log to
+      project: comet-examples-torchtune
+      experiment_name: my-experiment-name
+We automatically grab the config from the recipe you are running and log it to Comet. You can find it in the Comet Hyperparameters tab and the actual file in the :code:`Assets & Artifacts` tab.
+
+.. note::
+
+  Click on this sample `Comet project to see how it will looks like after fine-tuning <https://www.comet.com/examples/comet-example-torchtune-mistral/>`_.
+  The config used to train the models can be found `here <https://www.comet.com/examples/comet-example-torchtune-mistral/79dffc6e42424b33adfee6dd439ff47f?experiment-tab=params>`_.
```

Comment:
Looks like this is the original link with a run back in April? @ dzheng256 could you please demonstrate everything works okay by adding an example of a comet run using this PR?

@Lothiraldan feel free to chime in if you'd rather this run be on an official comet channel

Reply:
Happy to help, one benefit on having the example in our "officlal" workspace is that we can more easily update it when we make change to integrations so it stay up-to-date. But I don't have a strong opinion about it.

To be fully transparent, I was hoping to just restart the previous EC2 instance I used back in April but I am running into some quota issue. If @dzheng256 have more easy access to a training machine and is happy to run an experiment, we could get the PR merged more quickly. Again happy to go with whatever you recommend

Reply:
I have access to machines but only for netflix-internal inrfa. If you can set me up with the authentication for logging to the official workspace I could do that, but otherwise my runs are logged to our internal comet instance.